fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsFlightPhases.ts

```diff
@@ -0,0 +1,519 @@
+
+  memoLdgMemo_conf02: NXLogicConfirmNode;
+
+  memoLdgMemo_below2000ft: NXLogicMemoryNode;
+
+  memoToInhibit_conf01: NXLogicConfirmNode;
+
+  memoLdgInhibit_conf01: NXLogicConfirmNode;
+
+  previousTargetAltitude: number;
+
+  _wasBellowThreshold: boolean;
+
+  _wasAboveThreshold: boolean;
+
+  _wasInRange: boolean;
+
+  _wasReach200ft: boolean;
+
+  aircraft: Aircraft;
+
+  constructor(private fws: FwsCore) {
+    // momentary
+    this.toConfigTest = null;
+
+    // persistent
+    this.ldgMemo = null;
+    this.toMemo = null;
+
+    this.gndMemo = new NXLogicConfirmNode(1);
+
+    this.oneEngineRunningConf = new NXLogicConfirmNode(30);
+
+    this.speedAbove80KtsMemo = new NXLogicMemoryNode(true);
+
+    this.speedAboveV1Memo = new NXLogicMemoryNode();
+
+    this.mctMemo = new NXLogicConfirmNode(60, false);
+
+    this.firePBOutConf = new NXLogicConfirmNode(0.2);
+    this.firePBOutMemo = new NXLogicTriggeredMonostableNode(2);
+    this.firePBClear12 = new NXLogicMemoryNode(false);
+    this.phase112Memo = new NXLogicTriggeredMonostableNode(300);
+    this.phase10GroundMemo = new NXLogicTriggeredMonostableNode(2);
+    this.ac80KtsMemo = new NXLogicTriggeredMonostableNode(2);
+    this.prevPhase11InvertMemo = new NXLogicTriggeredMonostableNode(3, false);
+    this.twoEnginesTOPowerInvertMemo = new NXLogicTriggeredMonostableNode(1, false);
+    this.phase9Nvm = new NXLogicMemoryNode(true, true);
+    this.prevPhase11 = false;
+
+    this.groundImmediateMemo = new NXLogicTriggeredMonostableNode(2);
+    this.phase6Memo = new NXLogicTriggeredMonostableNode(15);
+    this.phase7Memo = new NXLogicTriggeredMonostableNode(120);
+    this.phase89Memo = new NXLogicTriggeredMonostableNode(180);
+
+    this.memoTo_conf01 = new NXLogicConfirmNode(120, true);
+    this.memoTo_memo = new NXLogicMemoryNode(false);
+
+    this.memoLdgMemo_conf01 = new NXLogicConfirmNode(1, true);
+    this.memoLdgMemo_inhibit = new NXLogicMemoryNode(false);
+    this.memoLdgMemo_conf02 = new NXLogicConfirmNode(10, true);
+    this.memoLdgMemo_below2000ft = new NXLogicMemoryNode(true);
+
+    this.memoToInhibit_conf01 = new NXLogicConfirmNode(3, true);
+
+    this.memoLdgInhibit_conf01 = new NXLogicConfirmNode(3, true);
+
+    // altitude warning
+    this.previousTargetAltitude = NaN;
+    this._wasBellowThreshold = false;
+    this._wasAboveThreshold = false;
+    this._wasInRange = false;
+    this._wasReach200ft = false;
+  }
+
+  update(deltaTime: number) {
+    this._updateFlightPhase(deltaTime);
+    this._updateTakeoffMemo(deltaTime);
+    this._updateLandingMemo(deltaTime);
+    this._updateAltitudeWarning();
+  }
+
+  _updateFlightPhase(_deltaTime: number) {
+    const raHeight1InValid = this.fws.radioHeight1.isFailureWarning() || this.fws.radioHeight1.isNoComputedData();
+    const raHeight2InValid = this.fws.radioHeight2.isFailureWarning() || this.fws.radioHeight2.isNoComputedData();
+    let radioHeight;
+    if (raHeight1InValid) {
```

Comment:
Is this meant to be invalid or in valid? I'm assuming its invalid so maybe the var naming should be `raHeight1Invalid` and `raHeight2Invalid`

Reply:
invalid, but this code was copied from the previous flight phase implementation